src/game-of-life/implementing.md

```diff
@@ -390,6 +390,109 @@ Make sure your development server is still running (run `npm run start` inside
+
+```rust
+#[wasm_bindgen]
+impl Universe { 
+    // ...
+
+    pub fn set_width(mut self, width: u32) -> Self {
+        self.width = width;
+        // Reinitialize cells with new width and the universe's height.
+        // Set all the cells to dead.
+        self.cells = (0..width * self.height).map(|_i| Cell::Dead).collect();
+        self
+    }
+
+    pub fn set_height(mut self, height: u32) -> Self {
+        self.height = height;
+        // Reinitialize cells with new height and the universe's width.
+        // Set all the cells to dead.
+        self.cells = (0..self.width * height).map(|_i| Cell::Dead).collect();
+        self
+    }
+
+}
+```
+
+Now we are going to make some public Rust functions for our `Universe`. We aren't going to compile them to WebAssembly, however. Rust compiled into WebAssembly functions cannot return borrowed references or take dynamically sized vectors as function arguments. Try compiling the Rust to WebAssembly by adding the `#[wasm_bindgen]` attribute and take a look at the errors you get.
+
+```rust
+pub fn cells(u: &Universe) -> &[Cell] {
+    &u.cells
+}
+
+/// Pass a vector of indices to set only those cells to be alive.
+pub fn set_cells(u: &mut Universe, indices: Vec<u32>) -> &mut Universe {
+    let input_cells: Vec<Cell> = (0..u.width * u.height)
+        .map(|i| {
+            if indices.contains(&i) {
+                Cell::Alive
+            } else {
+                Cell::Dead
+            }
+        })
+        .collect();
+    u.cells = input_cells;
+    u
+}
+```
+
+The last step for our test setup is to create a spaceship builder function. We'll want one for our input spaceship that we'll call the `tick` function on and we'll want the expected spaceship we will get after one tick. Since we're initializing a smaller 6 by 6 unit universe, the position of the spaceship before and after the tick was calculated manually. You can also confirm for yourself that the indices of the input spaceship after one tick are the same as the expected spaceship.
+
+```rust
+#[wasm_bindgen]
+pub fn input_spaceship() -> Universe {
+    let mut universe = Universe::new().set_width(6).set_height(6);
+    set_cells(&mut universe, vec![8, 15, 19, 20, 21]);
+    universe
+}
+
+#[wasm_bindgen]
+pub fn expected_spaceship() -> Universe {
+    let mut universe = Universe::new().set_width(6).set_height(6);
+    set_cells(&mut universe, vec![13, 15, 20, 21, 26]);
+    universe
+}
+```
+
+Now we're going to create our test in the `wasm_game_of_life/tests/web.rs` file.
+
+Before we do that, there is already one working test in the file. You can confirm that the Rust test functions compiled to WebAssembly are working by running `wasm-pack test --chrome --headless` in the `wasm-game-of-life` directory.
```

Comment:
Perhaps mention that you must have one of Chrome, Firefox, or Safari installed, and can run with `--firefox` and `--safari` as well?